src/python/pants/backend/python/goals/pytest_runner_integration_test.py

```diff
@@ -535,6 +535,7 @@ def test_bar():
                     dependencies=[":lib"],
                     provides=python_artifact(name="foo", version="9.8.7", setup_script="setup.py"),
                     sdist=False,
+                    generate_setup=False,
```

Comment:
@benjyw helped me with this offline. The current state of affairs is a footgun:
1. python_artifact setup_script is _not_ a setuptools argument, its an argument our rules look at sneakily even though python_artifact feigns **kwargs / don't care / pass-thru.
2. We don't consult setup_script's existence and do the right thing with generate_setup - we stomp that script and write our own.

Reply:
And, as such, you need to specify both setup_script=X on python_aritfact _and_ remember or even know to set generate_setup=False on the enclosing python_artifact target.

Reply:
Oh wait. Of course the issue here was PEBKAC. The `setup_script` argument was deprecated and I did not remove it from the test. One more diff coming.

Reply:
Still a bit of a footgun in that by default we'll overwrite your setup.py if it exists without the `generate_setup=False` being added.